flaml/nlp/autotransformers.py

```diff
@@ -0,0 +1,787 @@
+                    this_model = AutoModelForSequenceClassification.from_pretrained(checkpoint_path, config=model_config)
+            else:
+                this_model = AutoModelForSequenceClassification.from_pretrained(checkpoint_path, config=model_config)
+
+            this_model.resize_token_embeddings(len(self._tokenizer))
+            return this_model
+        else:
+            model_config = self._set_model_config(checkpoint_path, per_model_config, 1)
+            this_model = AutoModelForSequenceClassification.from_pretrained(checkpoint_path, config=model_config)
+            return this_model
+
+    def _get_metric_func(self):
+        if self.jobid_config.dat[0] in ("glue", "super_glue"):
+            metric = datasets.load.load_metric(self.jobid_config.dat[0], self.jobid_config.subdat)
+        elif self.jobid_config.dat[0] in ("squad", "squad_v2"):
+            metric = datasets.load.load_metric(self.jobid_config.dat[0])
+        else:
+            metric = datasets.load.load_metric(self.metric_name)
+        return metric
+
+    def _compute_metrics_by_dataset_name(self,
+                                         eval_pred):
+        predictions, labels = eval_pred
+        predictions = np.squeeze(predictions) \
+            if self.task_name == "regression" else np.argmax(predictions, axis=1)
+        metric_func = self._get_metric_func()
+        return metric_func.compute(predictions=predictions, references=labels)
+
+    def _compute_checkpoint_freq(self,
+                                 num_train_epochs,
+                                 batch_size,
+                                 mode="last"):
+        assert mode in {"last"}
+        if "gpu" in self._resources_per_trial:
+            ckpt_step_freq = int(min(num_train_epochs, 1) * len(self.train_dataset) / batch_size /
+                                 self._resources_per_trial["gpu"] / self.ckpt_per_epoch) + 1
+        else:
+            ckpt_step_freq = int(min(num_train_epochs, 1) * len(self.train_dataset) / batch_size /
+                                 self._resources_per_trial["cpu"] / self.ckpt_per_epoch) + 1
+
+        return ckpt_step_freq
+
+    @staticmethod
+    def _separate_config(config):
+        training_args_config = {}
+        per_model_config = {}
+
+        for key in config.keys():
+            if key in TrainingArguments.__dict__.keys():
+                training_args_config[key] = config[key]
+            else:
+                per_model_config[key] = config[key]
+
+        return training_args_config, per_model_config
+
+    def _objective(self, config, reporter, checkpoint_dir=None):
+        def model_init():
+            return self._load_model()
+        from transformers.trainer_utils import set_seed
+        set_seed(config["seed"])
+        np.random.seed(config["seed"])
+        torch.manual_seed(config["seed"])
+        torch.cuda.manual_seed(config["seed"])
+        np.random.seed(config["seed"])
+        random.seed(config["seed"])
```

Comment:
What's the reason setting all these seeds this way? Why to have both 373 and 376?